### PR TITLE
feat(devices): AMS-347 Sync working status when assets are re-created

### PR DESCRIPTION
**Pull request by charanm**
AMS does not delete assets, even when the client app (SCM) deletes it.
It does a soft delete. When the same device with Vendor ID, Serial
number is registered again the clent application expects a status sync.

### CODE REVIEW

File: app/Global.java [hunk 1]
```diff
@@ -43,104 +43,120 @@
...
-    @Override
-    public F.Promise<Result> onBadRequest(Http.RequestHeader request, String error) {
-        logMeterMetrics(request);
-        return F.Promise.pure(BaseController.prepareResult(Http.Status.BAD_REQUEST, request.getQueryString("callback"), error));
+  }
+
+  @Override
+  public void onStop(Application app) {
+    super.onStop(app);
+    if (app.configuration().getBoolean(ENABLE_CRON_ELECTION, false)) {
```
Comment (surajk94) [inline app/Global.java:70]: Default value while starting CronLeaderElection is true. Default value should be true here as well, else stop won't be called when default values are used

File: app/com/logistimo/services/DeviceService.java [hunk 2]
```diff
@@ -236,170 +237,174 @@ public DeviceCreationResponse createOrUpdateDevice(DeviceRegisterRequest deviceR
...
+            "{} while extracting device meta data from json {}, updating device {}, {} without meta data",
+            e.getMessage(), deviceRequest.meta, deviceRequest.vId, deviceRequest.dId, e);
+      }
+    }
+  }
 
-          //Updating temperature sensor details
-          if (deviceRequest.sns != null) {
-            createOrUpdateTemperatureSensors(device, deviceRequest);
+  private void updateDeviceUsers(Device device, List<String> ons, Integer assetUserType) {
```
Comment (surajk94) [inline app/com/logistimo/services/DeviceService.java:376]: "ons" variable can be renamed to assetUsers

File: app/Global.java [hunk 3]
```diff
@@ -43,104 +43,120 @@
 import play.mvc.Result;
 
 public class Global extends GlobalSettings {
-    private static final Logger.ALogger LOGGER = Logger.of(Global.class);
+  private static final Logger.ALogger LOGGER = Logger.of(Global.class);
+  private static final String ENABLE_CRON_ELECTION = "enable.cron.election";
+  private static final String ENABLE_METRICS = "enable.metrics";
```
Comment (surajk94) [inline app/Global.java:48]: these 2 configurations should be added in application.conf & application.conf.template

File: conf/application.conf [hunk 4]
```diff
@@ -140,6 +140,7 @@ activemq.url = "tcp://localhost:61616"
 
 # Cron Scheduler
 cron.zoo.path=/amsCronLeader
+enable.cron.election=true
```
Comment (surajk94) [inline conf/application.conf:143]: missed "enable.metrics" config parameter or is it not required here?

**Comment by charanm** [comment]
jarvis

**Comment by charanm** [comment]
jarvis

File: conf/application.conf (hunk 4 above)
Comment (charanm) [reply conf/application.conf:143]: This is not really required, this option only exists to disable it during tests.

**Pull request merged**